cpp/src/slice2cs/Gen.cpp

```diff
@@ -3307,21 +3302,15 @@ Slice::Gen::HelperVisitor::visitInterfaceDefStart(const InterfaceDefPtr& p)
     _out << sp << nl << "public static " << name
          << "Prx createProxy(Ice.Communicator communicator, string proxyString) =>";
     _out.inc();
-    _out << nl << "uncheckedCast(Ice.ObjectPrxHelper.createProxy(communicator, proxyString));";
+    _out << nl << "new " << name << "PrxHelper(Ice.ObjectPrxHelper.createProxy(communicator, proxyString));";
```

Comment:
createProxy is not calling a PrxHelper private constructor, which is much cleaner.